libbpf-tools/tcpconnect.c

```diff
@@ -0,0 +1,419 @@
+// SPDX-License-Identifier: GPL-2.0
+// Copyright (c) 2020 Anton Protopopov
+//
+// Based on tcpconnect(8) from BCC by Brendan Gregg
+#include <sys/resource.h>
+#include <arpa/inet.h>
+#include <argp.h>
+#include <signal.h>
+#include <limits.h>
+#include <unistd.h>
+#include <time.h>
+#include <bpf/bpf.h>
+#include "tcpconnect.h"
+#include "tcpconnect.skel.h"
+#include "trace_helpers.h"
+#include "map_helpers.h"
+
+#define warn(...) fprintf(stderr, __VA_ARGS__)
+
+const char *argp_program_version = "tcpconnect 0.1";
+const char *argp_program_bug_address = "<bpf@vger.kernel.org>";
+static const char argp_program_doc[] =
+	"\ntcpconnect: Count/Trace active tcp connections\n"
+	"\n"
+	"EXAMPLES:\n"
+	"    tcpconnect             # trace all TCP connect()s\n"
+	"    tcpconnect -t          # include timestamps\n"
+	"    tcpconnect -p 181      # only trace PID 181\n"
+	"    tcpconnect -P 80       # only trace port 80\n"
+	"    tcpconnect -P 80,81    # only trace port 80 and 81\n"
+	"    tcpconnect -U          # include UID\n"
+	"    tcpconnect -u 1000     # only trace UID 1000\n"
+	"    tcpconnect -c          # count connects per src, dest, port\n"
+	"    tcpconnect --C mappath # only trace cgroups in the map\n"
+	"    tcpconnect --M mappath # only trace mount namespaces in the map\n"
+	;
+
+static int get_int(const char *arg, int *ret, int min, int max)
+{
+	char *end;
+	long val;
+
+	errno = 0;
+	val = strtol(arg, &end, 10);
+	if (errno) {
+		warn("strtol: %s: %s\n", arg, strerror(errno));
+		return -1;
+	} else if (end == arg || val < min || val > max) {
+		return -1;
+	}
+	if (ret)
+		*ret = val;
+	return 0;
+}
+
+static int get_ints(const char *arg, int *size, int *ret, int min, int max)
+{
+	const char *argp = arg;
+	int max_size = *size;
+	char *end;
+	long val;
+
+	for (*size = 0; *size < max_size; ) {
+		errno = 0;
+		val = strtol(argp, &end, 10);
+		if (errno) {
+			warn("strtol: %s: %s\n", arg, strerror(errno));
+			return -1;
+		} else if (end == arg || val < min || val > max) {
+			return -1;
+		}
+		ret[*size] = val;
+		*size += 1;
+		if (*end == 0)
+			break;
+		argp = end+1;
```

Comment:
I noticed you did a lot of `end+1` and other cases were you have no spaces around arithmetic operator. I think kernel style dictates there should be spaces around them, and in BPF code we try to follow kernel code style, so would be nice to keep it consistent.

Reply:
Added spaces, I also prefer spaces, so I wonder where this came from. This is the only one place I see, btw